src/Common/RemoteProxyConfigurationResolver.cpp

```diff
@@ -16,22 +15,41 @@ namespace ErrorCodes
     extern const int BAD_ARGUMENTS;
 }
 
+std::string RemoteProxyHostFetcherImpl::fetch(const Poco::URI & endpoint, const ConnectionTimeouts & timeouts)
+{
+    auto request = Poco::Net::HTTPRequest(Poco::Net::HTTPRequest::HTTP_GET, endpoint.getPath(), Poco::Net::HTTPRequest::HTTP_1_1);
+    auto session = makeHTTPSession(HTTPConnectionGroupType::HTTP, endpoint, timeouts);
+
+    session->sendRequest(request);
+
+    Poco::Net::HTTPResponse response;
+    auto & response_body_stream = session->receiveResponse(response);
+
+    if (response.getStatus() != Poco::Net::HTTPResponse::HTTP_OK)
```

Comment:
Could be here redirects? Do we need to follow redirects here?

Reply:
I think it depends on the http server. Honestly, this "remote resolver" feature seems to be very unique for me and doesn't sound like a lot of people rely on it. I propose we keep it simple and implement it if the need arises.